pkg/engine/lifecycletest/pulumi_test.go

```diff
@@ -631,6 +631,35 @@ func TestProviderCancellation(t *testing.T) {
 	<-done
 }
 
+func TestLanguageRuntimeCancellation(t *testing.T) {
+	t.Parallel()
+
+	ctx, cancel := context.WithCancel(context.Background())
+
+	gracefulShutdown := false
+	programF := func() plugin.LanguageRuntime {
+		return deploytest.NewLanguageRuntimeWithShutdown(func(_ plugin.RunInfo, monitor *deploytest.ResourceMonitor) error {
+			time.Sleep(100 * time.Millisecond)
```

Comment:
Not sure if there's a better way to do this? But it seems I just needed to do an amount of waiting while stuff set up